WalletWasabi/WabiSabi/Backend/Rounds/Round.cs

```diff
@@ -115,6 +115,7 @@ public void SetPhase(Phase phase)
 
 	public void EndRound(EndRoundState finalState)
 	{
+		CoinjoinState = Assert<SigningState>().PublishWitnesses();
```

Comment:
This line assumes that the round is in the transaction signing phase, which is not generally true. This is likely the reason why the test are failing.

Reply:
After fix, the PublishWitness() is called only if CoinjoinState is of type SigningState.